server/blob/blob-api/src/main/java/org/apache/james/blob/api/ResolvedBucketName.java

```diff
@@ -0,0 +1,66 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.api;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Objects;
+
+public final class ResolvedBucketName {
```

Comment:
This POJO is defined in the API but only used in blob-s3.

I think introducing it is a good idea, but we should avoid poluting the API and move it into blob-s3.

Reply:
Hi @chibenwa ,
Do you have a plan to migrate to Java 14 (or 16), this will prevent from lot of code like this one.

Reply:
If we upgrade better a LTS version, so 17 (or even 21 would be better, but maybe too big of a jump?)

I would support such a migration, but I'm not sure the all community is ready to upgrade their java version :)

Could perhaps start a thread on the ML though?

Reply:
>Hi @chibenwa ,
> Do you have a plan to migrate to Java 14 (or 16), this will prevent from lot of code like this one.

`You` is out of place, as a community we should embrasse a `we`.

I would personnalyy support a switch to Java 17 or even 21.

Do you think you can pull off a mail on this topic on the mailing lists?